src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx

```diff
@@ -39,40 +57,73 @@ export const ConfigureTasks: React.FC<Props> = ({
   );
 
   const tasks = selectedBuildVariants.map(
-    (bv) => selectedBuildVariantTasks[bv]
+    (bv) => selectedBuildVariantTasks[bv] || {}
   );
 
   const currentTasks = deduplicateTasks(tasks);
+  const currentDownstreamPatches = getVisibleDownstreamPatches(
+    selectedDownstreamPatches,
+    selectedBuildVariants
+  );
+  const currentChildPatches = getVisibleChildPatches(
+    childPatches,
+    selectedBuildVariants
+  );
+
+  // Show details related to child patches (i.e. list all variants/tasks) only if it is the only menu item selected
+  const enumerateChildPatches =
+    currentChildPatches.length === 1 && selectedBuildVariants.length === 1;
 
   const onClickCheckbox = (taskName: string) => (e) => {
     const selectedBuildVariantsCopy = { ...selectedBuildVariantTasks };
-    selectedBuildVariants.forEach((bv) => {
-      if (selectedBuildVariantsCopy[bv][taskName] !== undefined) {
-        selectedBuildVariantsCopy[bv][taskName] = e.target.checked;
+    const selectedDownstreamPatchesCopy = { ...selectedDownstreamPatches };
+    selectedBuildVariants.forEach((v) => {
+      if (selectedBuildVariantsCopy?.[v]?.[taskName] !== undefined) {
+        selectedBuildVariantsCopy[v][taskName] = e.target.checked;
+      } else if (selectedDownstreamPatchesCopy?.[v] !== undefined) {
+        selectedDownstreamPatchesCopy[v] = e.target.checked;
       }
     });
     setSelectedBuildVariantTasks(selectedBuildVariantsCopy);
+    setSelectedDownstreamPatches(selectedDownstreamPatchesCopy);
   };
 
   const onClickSelectAll = (e) => {
     const selectedBuildVariantsCopy = { ...selectedBuildVariantTasks };
-    selectedBuildVariants.forEach((bv) => {
-      Object.keys(selectedBuildVariantsCopy[bv]).forEach((task) => {
-        selectedBuildVariantsCopy[bv][task] = e.target.checked;
-      });
+    const selectedDownstreamPatchesCopy = { ...selectedDownstreamPatches };
+    selectedBuildVariants.forEach((v) => {
+      if (selectedBuildVariantsCopy?.[v] !== undefined) {
+        Object.keys(selectedBuildVariantsCopy[v]).forEach((task) => {
+          selectedBuildVariantsCopy[v][task] = e.target.checked;
+        });
+      } else if (selectedDownstreamPatchesCopy?.[v] !== undefined) {
+        selectedDownstreamPatchesCopy[v] = e.target.checked;
+      }
     });
     setSelectedBuildVariantTasks(selectedBuildVariantsCopy);
+    setSelectedDownstreamPatches(selectedDownstreamPatchesCopy);
   };
 
-  const selectAllCheckboxState = getSelectAllCheckboxState(currentTasks);
-  const selectAllCheckboxCopy = `Select all tasks in ${
-    selectedBuildVariants.length > 1 ? "these variants" : "this variant"
-  }`;
+  const selectAllCheckboxState = getSelectAllCheckboxState(
+    currentTasks,
+    currentDownstreamPatches,
+    enumerateChildPatches
+  );
+  const selectAllCheckboxCopy =
+    Object.entries(currentTasks).length === 0
+      ? `Select all tasks in ${
```

Comment:
I'm personally a bigger fan of wording it "Add alias to patch" as opposed to "select all tasks in ___", because the user is not able to have any control over the tasks in the alias. But if you prefer to keep it like this and worry about it in the ticket that adds the tasks, that works too.